### DISCUSSION TITLE
Downgrade support

### DISCUSSION THREAD

**Discussion by zamazan4ik** [ory/keto#982] [Q&A] [answered]
Hi.

After reading the documentation (https://www.ory.sh/docs/keto/guides/upgrade) there are some open questions. Could you please describe somewhere in the documentation:

- Not clear, could I upgrade Keto with no downtime. Could not find a mention about it on this page
- How to downgrade Keto to the older version (hopefully with no downtime)

Thanks in advance!

**Comment by vinckr** [comment]
Hello @zamazan4ik 

>Not clear, could I upgrade Keto with no downtime. Could not find a mention about it on this page

Yes this is possible, but we do not have documentation on how to do upgrades without downtime at the moment. If you figure out a good solution, always open for contributions!

>How to downgrade Keto to the older version (hopefully with no downtime)

To be honest this is the first time someone asks for a downgrade, can you elaborate a bit on the reason for this? 
Otherwise I am not sure it is possible, there might be problems with the migrations - I would suggest to try it out and report here or in the chat if you run into issues, 
thanks!

**Reply by zamazan4ik** [reply]
> can you elaborate a bit on the reason for this

Sure! We are investigating the ways what should be done with Keto setup, if after an upgrade Keto setup will be broken. Having a possibility to perform a seamless downgrade with no interruption is a big advantage.

**Reply by vinckr** [reply, marked as answer]
I realize its quite late, but Ory offers support for self-hosting if you are still interested @zamazan4ik: https://www.ory.sh/support/
